packages/amplify_analytics_plugin_interface/lib/src/Analytics/AnalyticsUserProfile.dart

```diff
@@ -16,27 +16,27 @@
 import '../../amplify_analytics_plugin_interface.dart';
 
 class AnalyticsUserProfile {
-  String name;
-  String email;
-  String plan;
-  AnalyticsUserProfileLocation location;
-  AnalyticsProperties properties;
+  String? name;
+  String? email;
+  String? plan;
+  AnalyticsUserProfileLocation? location;
+  AnalyticsProperties? properties;
 
   AnalyticsUserProfile(
-      {String name,
-      String email,
-      String plan,
-      AnalyticsUserProfileLocation location,
-      AnalyticsProperties analyticsProperties}) {
+      {String? name,
+      String? email,
+      String? plan,
+      AnalyticsUserProfileLocation? location,
+      AnalyticsProperties? analyticsProperties}) {
     this.name = name;
     this.email = email;
     this.plan = plan;
     this.location = location;
     this.properties = analyticsProperties;
   }
 
-  Map<String, Object> getAllProperties() {
-    Map<String, Object> allProperties = {};
+  Map<String, Object?> getAllProperties() {
```

Comment:
Hey we do remove nulls from the map if they are null via this line:

  allProperties.removeWhere((_, v) => v == null);

The reason we do it this way is to make the code cleaner as otherwise we would need to add a lot more if/else statements to conditionally add in the values. 

For your second point from a minor optimization point of view it means less is beings send on the method channel from dart to native.  Also not sending the actual key "bar" changes how we do the check on the native side between checking if a key exists or checking if the value of a key is null.

Reply:
> Hey we do remove nulls from the map if they are null via this line
>
> allProperties.removeWhere((_, v) => v == null);

Yeah I did see this pattern in other places. But we're still doing the null checks in several places like in this file, for example:
```dart
if (name != null) {
  allProperties["name"] = name;
}
if (email != null) {
  allProperties["email"] = email;
}
```

Anyway, it's not a blocker. Just a suggestion on existing code

Reply:
Updated all patterns in code to use removeWhere